backend/src/disk/main.rs

```diff
@@ -267,27 +274,26 @@ pub async fn mount_fs<P: AsRef<Path>>(
         .arg(format!("{}_{}", guid, name))
         .invoke(crate::ErrorKind::DiskManagement)
         .await?;
-    mount(
-        Path::new("/dev/mapper").join(format!("{}_{}", guid, name)),
-        datadir.as_ref().join(name),
-        ReadWrite,
-    )
-    .await?;
+    let mapper_path = Path::new("/dev/mapper").join(format!("{}_{}", guid, name));
+    let reboot = e2fsck(&mapper_path, repair).await?;
+    mount(&mapper_path, datadir.as_ref().join(name), ReadWrite).await?;
 
     tokio::fs::remove_file(PASSWORD_PATH)
         .await
         .with_ctx(|_| (crate::ErrorKind::Filesystem, PASSWORD_PATH))?;
 
-    Ok(())
+    Ok(reboot)
 }
 
 #[instrument(skip(datadir, password))]
 pub async fn mount_all_fs<P: AsRef<Path>>(
     guid: &str,
     datadir: P,
+    repair: bool,
     password: &str,
-) -> Result<(), Error> {
-    mount_fs(guid, &datadir, "main", password).await?;
-    mount_fs(guid, &datadir, "package-data", password).await?;
-    Ok(())
+) -> Result<RequiresReboot, Error> {
+    let mut reboot = RequiresReboot::False;
+    reboot |= mount_fs(guid, &datadir, "main", repair, password).await?;
```

Comment:
jw what `|=` does?

Reply:
or-equals. Or the lhs with the rhs, then store the result in the lhs.

Reply:
Typescript I think has this now
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#short-circuiting-assignment-operators

https://github.com/tc39/proposal-logical-assignment